lib/yt/models/snippet.rb

```diff
@@ -51,6 +51,17 @@ def can_reply?
         @can_reply ||= data.fetch 'canReply', false
       end
 
+      def top_level_comment
+        if @top_level_comment.nil?
+          d = data.fetch('topLevelComment', nil).tap do |data|
+            data[:id] = data.fetch 'id', nil
+            data[:snippet] = data.fetch 'snippet', nil
+          end
+          @top_level_comment = Yt::Comment.new(d) if d
+        end
+        @top_level_comment
+      end
```

Comment:
There are many `if`s in this method. Are they all needed?
What happened if the code was as easy as the following?

```ruby
def top_level_comment
  @top_level_comment ||= Yt::Comment.new data['topLevelComment']
end
```

What side cases would we be missing?